anvio/drivers/prodigal.py

```diff
@@ -21,11 +26,108 @@
 __maintainer__ = "A. Murat Eren"
 __email__ = "a.murat.eren@gmail.com"
 
-
 run = terminal.Run()
 progress = terminal.Progress()
 pp = terminal.pretty_print
 
+# How many times to retry the gene calling.
+NUM_RETRIES = 3
+
+
+def run_command_wrapper(command, log_file_path, run, retries=NUM_RETRIES):
+    """A wrapper around `utils.run_command`.
+
+    It will retry the command `retries` time.  Sometimes commands can fail for weird reasons so retrying often fixes
+    things.
+
+    If the return value of the command is zero, then it is returned.  If not, an instance of CommandError is returned.
+    """
+    for attempt in range(retries):
+        try:
+            ret_val = utils.run_command(command, log_file_path)
+        except ConfigError as e:
+            # utils.run_command will raise ConfigError on certain types of failures.
+            run.warning(f'Caught error: {e}, while running command {command}, attempt {attempt}, '
+                        'but we are going to keep trying!')
+            continue
+
+        # Return code of zero means success in the unix world.
+        if ret_val == 0:
+            return ret_val
+
+    # Failed to run the command a couple of times, so return an error.
+    return CommandError(f'Failed to run the command after {retries} tries. Final exit code was: {ret_val}')
+
+
+def split_fasta_wrapper(fasta_file_path, num_splits):
```

Comment:
I think it would be much better to use the `utils.split_fasta` directly here.

If you think checking whether splits exist is essential, you can change the header of the existing function from,

```python
def split_fasta(input_file_path, parts=1, prefix=None):
```

to

``` python
def split_fasta(input_file_path, parts=1, prefix=None, check_splits=False):
    (...)

    if check_splits:
        (check  splits here)

    (...)
```

to check for splits :) But I am not sure if it is really necessary. by the way, this will probably explode if the FASTA file contains number of sequences that are smaller than the number of threads. please test for that to make sure it is OK :)

Reply:
Hmm, I wouldn't say checking the splits is essential...I suspect that an exception would be raised in the `split_fasta` function if there were a problem writing the sequences and closing the file.  It was more of sprinkling in assertions to catch things I'm assuming to be true.  However, there are also many checks that files exist later, so it could definitely be considered redundant!

Long winded way to say that I'll switch it to not use the wrapper.

Reply:
> this will probably explode if the FASTA file contains number of sequences that are smaller than the number of threads. please test for that to make sure it is OK :)

Looking at the `split_fasta` code, it looks like it handles this case ...see: https://github.com/merenlab/anvio/blob/849f998c8be3b4eabebefb485f0bd0d79e67e6a0/anvio/utils.py#L748

This looks like a good candidate for some unit tests though...I can add some to be sure it's working that way.

Reply:
Thank you! :)